presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -279,8 +280,8 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                 stats,
                 ImmutableSet.of(
                         new AddIntermediateAggregations(),
-                        new RemoveRedundantIdentityProjections())));
-
+                        new RemoveRedundantIdentityProjections(),
+                        new PruneCountAggregationOverScalar())));
```

Comment:
why is it added so late? I would add it add line 150 or so. The sooner this optimization is fired the better for other rules.